File: app/models/pay.py
import uuid
from app.extensions import db # 1단계에서 만든 db 도구를 가져옵니다.
from sqlalchemy.sql import func # 데이터베이스 자체의 현재 시간을 사용하기 위해 가져옵니다.

# Base 대신 db.Model을 상속받습니다.
class Pay(db.Model):
    __tablename__ = 'pay'

    pay_no = db.Column(db.BigInteger, primary_key=True, autoincrement=True)
    user_coupon_id = db.Column(db.BigInteger, db.ForeignKey('user_coupon.user_coupon_id'), nullable=True)
    reservation_key = db.Column(db.BigInteger, db.ForeignKey('reservation.reservation_key'), nullable=True)
    merchant_uid = db.Column(db.String(36), unique=True, nullable=False, default=lambda: str(uuid.uuid4()))
    pay_name = db.Column(db.String(255), nullable=False)
    pay_method = db.Column(db.String(50), nullable=True)
    buyer_name = db.Column(db.String(50), nullable=True)
    buyer_email = db.Column(db.String(50), nullable=True)
    pay_del_name = db.Column(db.String(50), nullable=True)
    pay_del_call = db.Column(db.String(50), nullable=True)
    pay_del_postcode = db.Column(db.String(5), nullable=True)
    pay_del_addr = db.Column(db.String(255), nullable=True)
    pay_status = db.Column(db.String(20), nullable=False, default='READY')
    pay_created_at = db.Column(db.TIMESTAMP, server_default=func.now())
    pay_updated_at = db.Column(db.TIMESTAMP, server_default=func.now(), onupdate=func.now())
    tx_id = db.Column(db.String(100), nullable=True)
    pay_fail_reason = db.Column(db.String(255), nullable=True)

File: src/main/java/com/ticketflow/entity/Pay.java
package com.ticketflow.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "pay")
@Getter @Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Pay {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "pay_no")
    private Long payNo;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_coupon_id")
    private UserCoupon userCoupon;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "reservation_key")
    private Reservation reservation;

    @Column(name = "merchant_uid", length = 36, unique = true, nullable = false)
    @Builder.Default
    private String merchantUid = UUID.randomUUID().toString();

    @Column(name = "pay_name", length = 255, nullable = false)
    private String payName;

    @Column(name = "pay_amount")
    private Long payAmount;

    @Column(name = "pay_method", length = 50)
    private String payMethod;

    @Column(name = "buyer_name", length = 50)
    private String buyerName;

    @Column(name = "buyer_email", length = 50)
    private String buyerEmail;

    @Column(name = "pay_del_name", length = 50)
    private String payDelName;

    @Column(name = "pay_del_call", length = 50)
    private String payDelCall;

    @Column(name = "pay_del_postcode", length = 5)
    private String payDelPostcode;

    @Column(name = "pay_del_addr", length = 255)
    private String payDelAddr;

    @Column(name = "pay_status", length = 20, nullable = false)
    @Builder.Default
    private String payStatus = "READY";

    @CreationTimestamp
    @Column(name = "pay_created_at", updatable = false)
    private LocalDateTime payCreatedAt;

    @UpdateTimestamp
    @Column(name = "pay_updated_at")
    private LocalDateTime payUpdatedAt;

    @Column(name = "pay_fail_reason", length = 255)
    private String payFailReason;

    @Column(name = "ls_order_id", length = 100)
    private String LsOrderId;

    @Column(name = "currency", length = 10)
    private String currency;

    @Column(name = "receipt_url",length = 500)
    private String receiptUrl;

    @Column(name = "ls_customer_id", length = 100)
    private String LsCustomerId;

    @Column(name = "ls_webhook_event_id",unique = true, length = 50)
    private String LsWebhookEventId;
}
